app/frontend/app/Http/Controllers/ProfileController.php

```diff
@@ -53,4 +55,31 @@ public function wishlist()
 
         return view('profile.wishlist', compact('user', 'items'));
     }
+
+    public function edit(String $username)
+    {   $profile = User::where('username', $username)->first();
+        $user = Auth::user();
+        return view('profile.edit', [ 'user' => $profile, 'roles' => User::ROLES]);
+    }
+
+    public function destroy(String $username)
+    {   $user = User::where('username', $username)->first();
+        $user->delete();
+        return redirect()
+            ->route('admin.users')
+            ->with('status', 'User deleted successfully');
+    }
+
+    public function save(Request $request, String $username)
+    {   $user = User::where('username', $username)->first();
```

Comment:
formatting as a whole is way off on the PR but styleci should fix it automatically when done